master/static/srv/command-entrypoint.sh

```diff
@@ -3,8 +3,19 @@
 source /run/determined/task-signal-handling.sh
 source /run/determined/task-logging-setup.sh
 
+if [ -z "$DET_PYTHON_EXECUTABLE" ]; then
+    export DET_PYTHON_EXECUTABLE="python3"
+fi
+if ! which "$DET_PYTHON_EXECUTABLE" >/dev/null 2>&1; then
+    echo "error: unable to find python3 as \"$DET_PYTHON_EXECUTABLE\"" >&2
+    echo "please install python3 or set the environment variable DET_PYTHON_EXECUTABLE=/path/to/python3" >&2
+    exit 1
+fi
+
 set -e
 
+"$DET_PYTHON_EXECUTABLE" -m determined.exec.prep_container --proxy
```

Comment:
doesn't have to be in this ticket, but the trial entrypoint, `entrypoint.sh` in this folder, suffers the same bug

Reply:
I had originally made a similar change in "entrypoint.sh", but when I ran an experiment, I got an error, so I pulled out the change.

```
<info> [2023-07-27 15:37:12] || INFO: Trial 152990 (Experiment 11277) was assigned to an agent
<info> [2023-07-27 15:37:20] [cont=0] || INFO: [1701104] root: detected 1 gpus
<info> [2023-07-27 15:37:20] [cont=0] || INFO: [1701104] root: Running task container on agent_id=node002, hostname=node002 with visible GPUs ['GPU-ee310a06-c7fb-f921-f137-d605913909b1']
<warning> [2023-07-27 15:37:21] [cont=0] || WARNING: [1701104] root: falling back to naive proxy ip resolution (error=None)
<none> [2023-07-27 15:37:22] [cont=0] || Traceback (most recent call last):
<none> [2023-07-27 15:37:22] [cont=0] ||   File "/opt/conda/lib/python3.8/runpy.py", line 194, in _run_module_as_main
<none> [2023-07-27 15:37:22] [cont=0] ||     return _run_code(code, main_globals, None,
<none> [2023-07-27 15:37:22] [cont=0] ||   File "/opt/conda/lib/python3.8/runpy.py", line 87, in _run_code
<none> [2023-07-27 15:37:22] [cont=0] ||     exec(code, run_globals)
<none> [2023-07-27 15:37:22] [cont=0] ||   File "/run/determined/pythonuserbase/lib/python3.8/site-packages/determined/exec/prep_container.py", line 358, in <module>
<none> [2023-07-27 15:37:22] [cont=0] ||     do_proxy(sess, info.allocation_id)
<none> [2023-07-27 15:37:22] [cont=0] ||   File "/run/determined/pythonuserbase/lib/python3.8/site-packages/determined/exec/prep_container.py", line 281, in do_proxy
<none> [2023-07-27 15:37:22] [cont=0] ||     set_proxy_address(sess, allocation_id)
<none> [2023-07-27 15:37:22] [cont=0] ||   File "/run/determined/pythonuserbase/lib/python3.8/site-packages/determined/exec/prep_container.py", line 265, in set_proxy_address
<none> [2023-07-27 15:37:22] [cont=0] ||     bindings.post_PostAllocationProxyAddress(
<none> [2023-07-27 15:37:22] [cont=0] ||   File "/run/determined/pythonuserbase/lib/python3.8/site-packages/determined/common/api/bindings.py", line 16610, in post_PostAllocationProxyAddress
<none> [2023-07-27 15:37:22] [cont=0] ||     _resp = session._do_request(
<none> [2023-07-27 15:37:22] [cont=0] ||   File "/run/determined/pythonuserbase/lib/python3.8/site-packages/determined/common/api/_session.py", line 36, in _do_request
<none> [2023-07-27 15:37:22] [cont=0] ||     return request.do_request(
<none> [2023-07-27 15:37:22] [cont=0] ||   File "/run/determined/pythonuserbase/lib/python3.8/site-packages/determined/common/api/request.py", line 180, in do_request
<none> [2023-07-27 15:37:22] [cont=0] ||     raise errors.APIException(r)
<none> [2023-07-27 15:37:22] [cont=0] || determined.common.api.errors.APIException: {"error":{"code":13,"reason":"Internal","error":"task.SetAllocationProxyAddress not supported for this allocation or resource manager"}}
<none> [2023-07-27 15:37:22] [cont=0] ||
<info> [2023-07-27 15:37:22] || INFO: Resources for Trial 152990 (Experiment 11277) have started
<error> [2023-07-27 15:38:34] || ERROR: Job failed in state FAILED. The job terminated with a non-zero exit code.
Job terminated. Most recent output from the stderr log is: INFO:    Using cached SIF image
WARNING: Overriding HOME environment variable with APPTAINERENV_HOME is not permitted INFO: Setting workdir to /run/determined/workdir
INFO: executing /run/determined/train/entrypoint.sh INFO: [1701104] root: detected 1 gpus
INFO: [1701104] root: Running task container on agent_id=node002, hostname=node002 with visible GPUs ['GPU-ee310a06-c7fb-f921-f137-d605913909b1'] WARNING: [1701104] root: falling back to naive proxy ip resolution (error=None)
Traceback (most recent call last):   File "/opt/conda/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,   File "/opt/conda/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)   File "/run/determined/pythonuserbase/lib/python3.8/site-packages/determined/exec/prep_container.py", line 358, in <module>
    do_proxy(sess, info.allocation_id)   File "/run/determined/pythonuserbase/lib/python3.8/site-packages/determined/exec/prep_container.py", line 281, in do_proxy
    set_proxy_address(sess, allocation_id)   File "/run/determined/pythonuserbase/lib/python3.8/site-packages/determined/exec/prep_container.py", line 265, in set_proxy_address
    bindings.post_PostAllocationProxyAddress(   File "/run/determined/pythonuserbase/lib/python3.8/site-packages/determined/common/api/bindings.py", line 16610, in post_PostAllocationProxyAddress
    _resp = session._do_request(   File "/run/determined/pythonuserbase/lib/python3.8/site-packages/determined/common/api/_session.py", line 36, in _do_request
    return request.do_request(   File "/run/determined/pythonuserbase/lib/python3.8/site-packages/determined/common/api/request.py", line 180, in do_request
    raise errors.APIException(r) determined.common.api.errors.APIException: {"error":{"code":13,"reason":"Internal","error":"task.SetAllocationProxyAddress not supported for this allocation or resource manager"}}
 srun: error: node002: task 0: Exited with exit code 1
srun: Terminating StepId=174.0 
```

Reply:
thanks for the stack. i'll file a ticket for my team.